### PR TITLE
47342_Trainee_view_course_members

### PR DESCRIPTION
**Pull request by Omid07**
screenshot from 2015-12-11 12 20 40

### CODE REVIEW

File: app/Http/Controllers/Supervisor/CourseUsersController.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function store(Request $request, $courseId)
         User::whereIn('id', $detachIds)->update(['is_enrolled' => 0]);
         User::whereIn('id', $traineeIds)->update(['is_enrolled' => 1]);
         $course->users()->attach($traineeIds, ['is_currently_enrolled' => 1]);
-        $course->users()->detach($detachIds);
+        // $course->users()->detach($detachIds);
```
Comment (asifbuetcse) [inline app/Http/Controllers/Supervisor/CourseUsersController.php]: comment out?

File: resources/views/trainee/users/index.blade.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                    <td>
+                        {!! Html::linkAction('Trainee\UsersController@show', 'View', ['memberId' => $member->id]) !!}
+                    </td>
+                </tr>
+            @endforeach
+        </tbody>
+    </table>
+</div>
+<center>
+
```
Comment (asifbuetcse) [inline resources/views/trainee/users/index.blade.php]: space

**Comment by asifbuetcse** [comment]
ok

File: app/Http/Controllers/Trainee/UsersController.php [hunk 3]
```diff
@@ -17,7 +18,11 @@ class UsersController extends Controller
      */
     public function index()
     {
-        //
+        $id = \Auth::user()->id;
+        $courseHistoryId = CourseUser::where('user_id', $id)->where('is_currently_enrolled', 1)->lists('course_id');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]: Use Eloquent Relationship

File: app/Http/Controllers/Trainee/UsersController.php [hunk 4]
```diff
@@ -17,7 +18,11 @@ class UsersController extends Controller
...
+        $traineesId = CourseUser::whereIn('course_id', $courseHistoryId)->where('is_currently_enrolled', 1)->lists('user_id');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]: Use Eloquent Relationship. Not where in

File: app/Http/Controllers/Trainee/UsersController.php [hunk 5]
```diff
@@ -17,7 +19,11 @@ class UsersController extends Controller
      */
     public function index()
     {
-        //
+        $id = \Auth::user()->id;
+        $courseId = CourseUser::whereUserId($id)->whereIsCurrentlyEnrolled(1)->first()->course_id;
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]: To get course. can you try
```php
$course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
```

File: app/Http/Controllers/Trainee/UsersController.php [hunk 6]
```diff
@@ -17,7 +19,11 @@ class UsersController extends Controller
      */
     public function index()
     {
-        //
+        $id = \Auth::user()->id;
+        $course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
+        $traineesId = Course::find($course->id)->users()->where('course_user.is_currently_enrolled', 1)->lists('user_id');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]:
```php
Course::find($course->id)
```
You get `$course`, and you find course AGAIN by it's id ??
Please take care of this.

File: app/Http/Controllers/Trainee/UsersController.php [hunk 7]
```diff
@@ -17,7 +19,10 @@ class UsersController extends Controller
      */
     public function index()
     {
-        //
+        $courseId = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first()->id;
+        $traineesId = Course::find($courseId)->users()->where('course_user.is_currently_enrolled', 1)->lists('user_id');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]: The first row: ```...->first()``` returns `$course`
Use this `$course` to find `userId`, don't find `course` again.

File: app/Http/Controllers/Trainee/UsersController.php [hunk 8]
```diff
@@ -17,7 +19,10 @@ class UsersController extends Controller
      */
     public function index()
     {
-        //
+        $course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
+        $traineesId = $course->users()->where('course_user.is_currently_enrolled', 1)->lists('user_id');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/UsersController.php]: You find all member
```php
$course->users()->where('course_user.is_currently_enrolled', 1)
```
and find list of id.
```php
lists('user_id')
```
Then you get members AGAIN
```php
ser::whereIn('id', $traineesId)
```
Please fix it.

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**